modules/logwriter/variables.tf

```diff
@@ -0,0 +1,84 @@
+variable "name" {
+  type        = string
+  description = <<-EOF
+    Name of role. Since this name must be unique within the
+    account, it will be reused for most of the resources created by this
+    module.
+  EOF
+
+  validation {
+    condition     = length(var.name) <= 64
+    error_message = "Name must be under 64 characters."
+  }
+}
+variable "bucket_arn" {
+  description = "S3 Bucket ARN to write log records to."
+  type        = string
+  nullable    = false
+}
+
+variable "prefix" {
+  description = "Optional prefix to write log records to."
+  type        = string
+  nullable    = false
+  default     = "observe"
+}
+
+variable "log_group_name_patterns" {
+  description = "List of patterns as strings. We will only subscribe to log groups that have names matching one of the provided strings based on strings based on a case-sensitive substring search. To subscribe to all log groups, use the wildcard operator *."
+  type        = list(string)
+  nullable    = true
+  default     = [""]
+}
+
+variable "log_group_name_prefixes" {
+  description = "List of prefixes as strings. The lambda function will only apply to log groups that start with a provided string. To subscribe to all log groups, use the wildcard operator *."
+  type        = list(string)
+  nullable    = true
```

Comment:
This should not be nullable, and the default should be `[]`.

You might also want to validate that the prefixes are valid. A "prefix" can either be:
- `*`, which denotes "any"
- follow the log group name restrictions
```
Log group names can be between 1 and 512 characters long. Log group names consist of the following characters: a-z, A-Z, 0-9, '_' (underscore), '-' (hyphen), '/' (forward slash), '. ' (period), and '#' (number sign) Log group names can't start with the string aws/
```